kart/status.py

```diff
@@ -96,14 +103,28 @@ def status(ctx, output_format):
         jdict["conflicts"] = conflicts_writer.list_conflicts()
         jdict["state"] = "merging"
     else:
-        jdict["workingCopy"] = get_working_copy_status_json(repo)
+        if list_untracked_tables:
+            is_untracked_tables_call = True
+            """Check for any untracked tables in working copy"""
+            wc = repo.working_copy.tabular 
+            if wc is not None:
+                if wc.session() is not None:     
+                    with wc.session() as sess:
+                        wc_items = wc.adapter.list_tables(sess)
+                    # Get all tables in working copy
+                    all_tables = [table_name for table_name, title in wc_items.items()]
+                    # Get tables shown in kart data ls
+                    datasets_paths = [dataset.table_name for dataset in repo.datasets(filter_dataset_type="table")] 
+                    # Get untracked tables
+                    untracked_tables = list(set(all_tables) - set(datasets_paths))
+                    jdict["untrackedTables"] = untracked_tables
+        jdict["workingCopy"] = get_working_copy_status_json(repo, untracked_tables, is_untracked_tables_call)
```

Comment:
The code here is fine, but it is in a strange order - it could be clearer if it was moved around.

Here is a nice way to structure code:
```
def do_big_task():
  result = ...
  result += do_first_small_task()
  result += do_second_small_task()
  if necessary:
    result += do_third_small_task()
```

The code right now however looks a bit more like this:
```
if necessary:
  third_small_task = doThirdSmallTask()
do_big_task(third_small_task=third_small_task)
```

So the big task you are doing is `get_working_copy_status_json`, and the smaller subtask you are doing within that is finding the untracked tables. `get_working_copy_status_json` could take a bool which controls whether or not to add the untracked tables. Then actually finding the untracked tables could be a subtask of that big task. Does that make sense?